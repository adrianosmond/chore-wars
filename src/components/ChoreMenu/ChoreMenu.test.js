import React from 'react';
import { shallow } from 'enzyme';
import { ChoreMenu } from './index';

describe('Chore Menu', () => {
  const mockRemoveChore = jest.fn();
  const shallowChoreMenu = shallow(<ChoreMenu slug={'test-chore'} removeChore={mockRemoveChore} />);
  const toggleButton = shallowChoreMenu.find('.js-toggle-menu');
  it('Renders a new chore menu', () => {
    expect(shallowChoreMenu).toMatchSnapshot();
  });

  it('Can be toggled', () => {
    toggleButton.simulate('click');
    expect(shallowChoreMenu.state().visible).toBe(true);
    expect(shallowChoreMenu).toMatchSnapshot();
    toggleButton.simulate('click');
    expect(shallowChoreMenu.state().visible).toBe(false);
  });

  it('Can remove a chore', () => {
    toggleButton.simulate('click');
    const removeButton = shallowChoreMenu.find('.js-remove-chore');
    removeButton.simulate('click');
    expect(mockRemoveChore).toHaveBeenCalled();
  });
});
