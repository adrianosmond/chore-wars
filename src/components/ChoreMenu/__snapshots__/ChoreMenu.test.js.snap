// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chore Menu Can be toggled 1`] = `
<div
  className="chore-menu"
>
  <button
    className="chore-menu__button js-toggle-menu"
    onClick={[Function]}
  >
    <span
      className="chore-menu__button-icon"
    >
      …
    </span>
  </button>
  <div
    className="chore-menu__menu chore-menu__menu--visible"
  >
    <Link
      className="chore-menu__item"
      replace={false}
      to="/chores/edit/test-chore"
    >
      Edit
    </Link>
    <button
      className="chore-menu__item js-remove-chore"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;

exports[`Chore Menu Renders a new chore menu 1`] = `
<div
  className="chore-menu"
>
  <button
    className="chore-menu__button js-toggle-menu"
    onClick={[Function]}
  >
    <span
      className="chore-menu__button-icon"
    >
      …
    </span>
  </button>
  <div
    className="chore-menu__menu"
  >
    <Link
      className="chore-menu__item"
      replace={false}
      to="/chores/edit/test-chore"
    >
      Edit
    </Link>
    <button
      className="chore-menu__item js-remove-chore"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;
